src/FunderMaps.Infrastructure/Storage/SpacesBlobStorageService.cs

```diff
@@ -189,26 +194,109 @@ public async Task StoreDirectoryAsync(string directoryName, string directoryPath
+                    e.UploadRequest.Headers.ContentType = storageObject?.ContentType ?? e.UploadRequest.Headers.ContentType;
+                    e.UploadRequest.Headers.CacheControl = storageObject?.CacheControl ?? e.UploadRequest.Headers.CacheControl;
+                    e.UploadRequest.Headers.ContentDisposition = storageObject?.ContentDisposition ?? e.UploadRequest.Headers.ContentDisposition;
+                    e.UploadRequest.Headers.ContentEncoding = storageObject?.ContentEncoding ?? e.UploadRequest.Headers.ContentEncoding;
+                };
+
+                TransferUtilityConfig config = new()
+                {
+                    /** Note: This is currently set to the default value of 10. I did some benchmarking on 23 dec 2020 
+                           and discovered that turning this value up will cause some unstable behaviour resulting in
+                           the task throwing an exception and being cancelled. Setting this to 20 seemed to work fine, 
+                           however 50 or above seems to result in crashes. I've left it on 10 for now to be safe for use 
+                           in production. Worth investigating later for a major performance increase!
+
+                           - Patrick
+                    **/
+                    ConcurrentServiceRequests = ConcurrentServiceRequests
+                };
+
+                await new TransferUtility(client, config).UploadDirectoryAsync(request);
+            }
+            catch (AmazonS3Exception e)
+            {
+                _logger.LogError(e, "Could not store directory to Spaces using S3");
+
+                throw new StorageException("Could not store directory", e);
+            }
+        }
+
+        /// <summary>
+        ///     Removes a directory.
+        /// </summary>
+        /// <param name="directoryPath">Full path of the directory to delete.</param>
+        /// <returns>See <see cref="ValueTask"/>.</returns>
+        public async Task RemoveDirectoryAsync(string directoryPath)
+        {
+            try
+            {
+                // NOTE: This call returns max. up to 1000 records by design
+                //       In order to obtain every record that matches our query, 
+                //       we have to repeat our request execution in a loop - using continuation tokens -- until no longer truncated.
+                //       See https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/S3/MS3ListObjectsV2AsyncListObjectsV2RequestCancellationToken.html
+                ListObjectsV2Request request = new()
+                {
+                    BucketName = _options.BlobStorageName,
+                    Prefix = directoryPath,
+                    MaxKeys = MaxKeys
+                };
+
+                List<Task> tasklist = new();
+
+                ListObjectsV2Response response;
+                do
```

Comment:
This should really be a for(..). There is an initializer `await client.ListObjectsV2Async(request);`, an advance `request.ContinuationToken = response.NextContinuationToken;` and a test `response.IsTruncated`